packages/eslint-config/index.js

```diff
@@ -21,14 +22,6 @@ module.exports = {
   ],
 
   overrides: [
-    {
-      files: ['*.ts', '*.tsx'],
-      rules: {
-        // These off-by-default or configurable rules are good and we like having them on
-        '@typescript-eslint/non-nullable-type-assertion-style': 'error',
-        '@typescript-eslint/prefer-optional-chain': 'error',
-      },
-    },
```

Comment:
Moved to specifically the `typescript.js` file since they're specific to TypeScript.